runtime/drivers/druid/druid.go

```diff
@@ -256,12 +257,19 @@ func (c *connection) AsObjectStore() (drivers.ObjectStore, bool) {
 
 // AsModelExecutor implements drivers.Handle.
 func (c *connection) AsModelExecutor(instanceID string, opts *drivers.ModelExecutorOptions) (drivers.ModelExecutor, bool) {
+	if opts.OutputHandle == c && opts.InputConnector == "gcs" {
```

Comment:
Connector names can be aliased (e.g. if you connect with two different service accounts for different buckets). So instead of checking `InputConnector` it should check `InputHandle.Driver()` instead.